libdnf/utils/utils.hpp

```diff
@@ -135,109 +158,86 @@ inline GoalProblem operator&(GoalProblem lhs, GoalProblem rhs) {
         static_cast<std::underlying_type<GoalProblem>::type>(rhs));
 }
 
-inline bool GoalJobSettings::get_strict(const libdnf::ConfigMain & cfg_main, bool set_used_value) {
+inline bool GoalJobSettings::resolve_strict(const libdnf::ConfigMain & cfg_main) {
+    auto resolved = GoalUsedSetting::UNUSED;
     switch (strict) {
         case GoalSetting::AUTO: {
             bool strict = cfg_main.strict().get_value();
-            if (set_used_value) {
-                if (used_strict != GoalUsedSetting::UNUSED) {
-                    throw LogicError("Used value for 'strict' already set");
-                }
-                used_strict = strict ? GoalUsedSetting::USED_TRUE : GoalUsedSetting::USED_FALSE;
-            }
-            return strict;
-        }
+            resolved = strict ? GoalUsedSetting::USED_TRUE : GoalUsedSetting::USED_FALSE;
+        } break;
         case GoalSetting::SET_TRUE:
-            if (set_used_value) {
-                if (used_strict != GoalUsedSetting::UNUSED) {
-                    throw LogicError("Used value for 'strict' already set");
-                }
-                used_strict = GoalUsedSetting::USED_TRUE;
-            }
-            return true;
+            resolved = GoalUsedSetting::USED_TRUE;
+            break;
         case GoalSetting::SET_FALSE:
-            if (set_used_value) {
-                if (used_strict != GoalUsedSetting::UNUSED) {
-                    throw LogicError("Used value for 'strict' already set");
-                }
-                used_strict = GoalUsedSetting::USED_FALSE;
-            }
-            return false;
+            resolved = GoalUsedSetting::USED_FALSE;
+            break;
     }
-    throw LogicError("Invalid 'best' value in GoalJobSettings");
+    if (resolved == GoalUsedSetting::UNUSED) {
+        throw LogicError("Invalid 'strict' value in GoalJobSettings");
+    }
+    if (used_strict != GoalUsedSetting::UNUSED && resolved != used_strict) {
+        throw LogicError("Used value for 'strict' already set");
```

Comment:
A more accurate report? Eg:
"GoalJobSettings::resolve_strict: 'strict' is already set to a different value"